### PR TITLE
Defined get_bgp_neighbors_detail() for JunOS & IOS-XR

### PR DESCRIPTION
**Pull request by mirceaulinic**
Child of #154

### CODE REVIEW

**Comment by dbarrosop** [comment]
Awesome one! : )

Could you fix the conflicts, please?

**Comment by dbarrosop** [comment]
We got new conflicts after I merged #176

**Comment by mirceaulinic** [comment]
Yes, after each merge will be like that because files do not include the previous method. In this branch for example junos.py does not include ```get_bgp_config``` previously merged etc. Will fix that in a bit.

**Comment by mirceaulinic** [comment]
Fixed conflicts & catch specific exception for EOS in get_bgp_config (did not have time to react, sorry) :)

File: test/unit/models.py [hunk 1]
```diff
@@ -118,4 +118,37 @@
     'local_as'          : int,
     'peer_as'           : int,
     'prefix_limit'      : dict
+
```
Comment (dbarrosop) [inline test/unit/models.py]: there is a missing `}` here

**Comment by mirceaulinic** [comment]
I'm impressed how fast you saw it

**Comment by mirceaulinic** [comment]
I will update the documentation too

**Comment by dbarrosop** [comment]
I have my ways ;)

Just kidding, just pulling your branch and running the unittests : )

Now it works but the unittest seems to be failing:

```
(napalm) ➜  unit git:(mirceaulinic-CF-BGP) nosetests -v TestJunOSDriver:TestGetterJunOSDriver.test_bgp_neighbors_detail
test_bgp_neighbors_detail (TestJunOSDriver.TestGetterJunOSDriver) ... FAIL

======================================================================
FAIL: test_bgp_neighbors_detail (TestJunOSDriver.TestGetterJunOSDriver)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/dbarroso/workspace/napalm/test/unit/base.py", line 254, in test_bgp_neighbors_detail
    self.assertTrue(result)
AssertionError: False is not true

----------------------------------------------------------------------
Ran 1 test in 0.003s

FAILED (failures=1)
```

**Pull request merged**